src/kademlia/routing_table.cpp

```diff
@@ -682,13 +701,37 @@ routing_table::add_node_status_t routing_table::add_node_impl(node_entry e)
 		else
 		{
 			TORRENT_ASSERT(existing->id != e.id);
-			// this is the same IP and port, but with
-			// a new node ID. remove the old entry and
-			// replace it with this new ID
+			// This is the same IP and port, but with a new node ID.
+			// This may indicate a malicious node so remove the entry.
+#ifndef TORRENT_DISABLE_LOGGING
+			char hex_id_new[41];
+			char hex_id_old[41];
+			aux::to_hex(reinterpret_cast<char const*>(&e.id[0]), 20, hex_id_new);
```

Comment:
if you call e.id.data(), you don't need the reinterpret cast. hopefully at some point I'll make all different parts agree on what the type of a "byte" is.